animation.h

```diff
@@ -3,13 +3,24 @@
 
 #include "location.h"
 
-enum class AnimationType { kDefault, kRunRight, kRunLeft };
+enum class AnimationType {
+  kDefault,
+  kRunRight,
+  kRunLeft,
+  kSSTransformation,
+  kSSIdle,
+  kSSRunRight,
+  kSSRunLeft
+};
 
 // A wrapper struct holding all the info we need about an animation.
 struct Animation {
   AnimationType type;
   int time_between_frames_ms;
 
+  // Offsets for each frame.
+  std::vector<Location> offsets;
```

Comment:
This is similar to having two maps for animation information, and the solution is the same. Associated data should be grouped together: make a struct that holds an SDL_Rect and a Location so that there isn't an implicit association with the two separate vectors.